.github/workflows/ci.yaml

```diff
@@ -0,0 +1,185 @@
+name: EC2 Metadata Mock CI and Release
+
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  DEFAULT_GO_VERSION: ^1.15
+  IS_RELEASE: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags/') }}
+  IS_PUSH: ${{ github.event_name == 'push' }}
+  # Skip running this stage when updating release versions as part of release prep
+  # as the new Docker image wouldn't be available yet and previous commits should have already been tested.
+  # Since the release prep commit will be the release commit, always run this stage when a new tag is pushed, as part of a release.
+  SHOULD_TEST_HELM: ${{ !contains(github.event.head_commit.message, 'Skip Helm E2E Tests') || (github.event_name == 'push' && contains(github.ref, 'refs/tags/')) }}
+  GITHUB_USERNAME: ${{ secrets.EC2_BOT_GITHUB_USERNAME }}
+  GITHUB_TOKEN: ${{ secrets.EC2_BOT_GITHUB_TOKEN }}
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+  DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+
+jobs:
+
+  buildAndTest:
+    name: Build and Test
+    runs-on: ubuntu-20.04
```

Comment:
any particular reason why we're locking in this ubuntu version vs. running on `ubuntu-latest`?

Reply:
I wanted this to be deterministic, I'm not opposed to using latest, but that was the reasoning.